DataWorkerService/BackgroundServices/TournamentProcessorService.cs

```diff
@@ -48,46 +51,72 @@ protected override async Task DoWork(IServiceScope scope, CancellationToken stop
-
-                logger.LogWarning("Skipping processing for tournament with no matches [Id: {Id}]", tournament.Id);
+                RejectTournamentIncompleteData(tournament);
                 continue;
             }
 
-            foreach (IProcessor<Tournament> processor in tournamentProcessorResolver.GetAll())
-            {
-                await processor.ProcessAsync(tournament, stoppingToken);
-            }
+            tasks.Add(ProcessAsync(tournament, tournamentProcessorResolver, stoppingToken));
+        }
+
+        await Task.WhenAll(tasks);
+        await context.SaveChangesAsync(stoppingToken);
+
+        _stopwatch.Stop();
+        logger.LogInformation(
+            @"Tournament batch processing completed: [Count: {Count}, Elapsed: {Elapsed:mm\:ss\:fff}]",
+            tournaments.Count, _stopwatch.Elapsed);
 
-            _stopwatch.Stop();
+        _stopwatch.Reset();
+    }
+
+    private void RejectTournamentIncompleteData(Tournament tournament)
+    {
+        tournament.VerificationStatus = VerificationStatus.Rejected;
+        tournament.RejectionReason = TournamentRejectionReason.IncompleteData;
+        tournament.ProcessingStatus = TournamentProcessingStatus.Done;
+        tournament.LastProcessingDate = DateTime.Now;
+
+        logger.LogWarning("Skipping processing for tournament with no matches [Id: {Id}]", tournament.Id);
+    }
+
+    private async Task ProcessAsync(Tournament tournament,
+        ITournamentProcessorResolver tournamentProcessorResolver, CancellationToken stoppingToken)
+    {
+        TournamentProcessingStatus processingStatusBefore = tournament.ProcessingStatus;
+        logger.LogDebug(
+            "Processing starting [Id: {Id} | Processing Status: {Status} | Last Processing: {LastProcessing:MM/dd/yyyy}]",
+            tournament.Id,
+            tournament.ProcessingStatus,
+            tournament.LastProcessingDate
+        );
+
+        IProcessor<Tournament> processor = tournamentProcessorResolver.GetNextProcessor(tournament.ProcessingStatus);
+        await processor.ProcessAsync(tournament, stoppingToken);
+
+        if (processingStatusBefore != tournament.ProcessingStatus)
+        {
             logger.LogInformation(
-                "Processing completed [Id: {Id} | Processing Status: {Status} | Elapsed: {Elapsed:mm\\:ss\\:fff}]",
+                "Processing completed [Id: {Id} | Processing Status: {Before} --> {After}]",
                 tournament.Id,
-                tournament.ProcessingStatus,
-                _stopwatch.Elapsed
+                processingStatusBefore,
+                tournament.ProcessingStatus
+            );
+        }
+        else
+        {
+            logger.LogDebug(
+                "Processing completed [Id: {Id} | Processing Status: {Before} --> {After}]",
+                tournament.Id,
+                processingStatusBefore,
+                tournament.ProcessingStatus
             );
```

Comment:
Not really what I meant LOL... it's the same log statement with the level changed

Reply:
```cs
            logger.LogDebug(
                "Processing completed [Id: {Id} | Processing Status: {Before} (No change)]",
                tournament.Id,
                processingStatusBefore
            );
```

Reply:
Fixed